lib/hexpm_web/views/view_helpers.ex

```diff
@@ -354,6 +354,57 @@ defmodule HexpmWeb.ViewHelpers do
     |> EQRCode.encode()
     |> EQRCode.svg(width: 250)
   end
+
+  # assumes positive values only, and graph dimensions of 800 x 200
+  def time_series_graph(points) do
+    max = Enum.max(points ++ [5])
+
+    y_axis_labels = y_axis_labels(0, max)
+
+    calculated_points =
+      points
+      |> Enum.map(fn p -> points_to_graph(max, p) end)
+      |> Enum.zip(x_axis_points(length(points)))
+
+    polyline_points = to_polyline_points(calculated_points)
+    polyline_fill = to_polyline_fill(calculated_points)
+
+    {y_axis_labels, polyline_points, polyline_fill}
+  end
+
+  defp points_to_graph(max, data) do
+    px_per_point = 200 / max
+    200 - (data |> Kernel.*(px_per_point) |> Float.round(3))
+  end
+
+  defp x_axis_points(total_points) do
+    # width / points captured
+    px_per_point = Float.round(800 / total_points, 2)
+    Enum.map(0..total_points, &Kernel.*(&1, px_per_point))
+  end
+
+  defp to_polyline_points(list) do
+    Enum.reduce(list, "", fn {y, x}, acc -> acc <> "#{x}, #{y} " end)
```

Comment:
Can you use `Enum.join` here?

Reply:
no, i dont think so. I need to reverse the tuple, and also need to add a comma between the x and y, which i dont think join can do.

Reply:
I was thinking mainly to introduce the comma, but since we are doing so many other things the reduce is probably better.